packages/lib/EventManager.ts

```diff
@@ -643,10 +648,34 @@ export default class EventManager {
           const bookingCalendarReference = booking.references.find((reference) =>
             reference.type.includes("_calendar")
           );
-          // There was a case that booking didn't had any reference and we don't want to throw error on function
-          if (bookingCalendarReference) {
-            // Update all calendar events.
+
+          if (bookingCalendarReference && bookingCalendarReference.uid) {
+            // Update existing calendar events
             results.push(...(await this.updateAllCalendarEvents(evt, booking, newBookingId)));
+          } else {
+            log.debug("No valid calendar event found for booking, creating new calendar events");
+            const createdEvents = await this.createAllCalendarEvents(evt);
+            results.push(...createdEvents);
+
+            // Update booking references with newly created calendar events
+            const calendarReferences = createdEvents
+              .filter((result) => result.type.includes("_calendar") && result.success)
+              .map((result) => ({
+                type: result.type,
+                uid: result.uid,
+                meetingId: result.uid,
+                meetingPassword: result.originalEvent.attendees?.[0]?.timeZone || "",
+                meetingUrl: result.originalEvent.videoCallData?.url || "",
+                externalCalendarId: result.externalId || null,
+                credentialId: result.credentialId || null,
+              }));
```

Comment:
The _reschedule_ fn is already doing a lot of things violating Single Responsibility principle. Let's do this handling in createMissingCalendarEvents or handleMissingCalendarEvents